src/net_processing.cpp

```diff
@@ -1371,15 +1370,16 @@ void PeerManagerImpl::MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid)
             lNodesAnnouncingHeaderAndIDs.push_back(nodeid);
             return;
         }
-        CNodeState *state = State(*it);
-        if (state != nullptr && !state->m_is_inbound) ++num_outbound_hb_peers;
+        PeerRef peer{GetPeerRef(*it)};
+        if (peer && !peer->m_is_inbound) ++num_outbound_hb_peers;
     }
-    if (nodestate->m_is_inbound) {
+    PeerRef peer{GetPeerRef(nodeid)};
```

Comment:
I moved it up in [07f4ceb](https://github.com/bitcoin/bitcoin/pull/30233/commits/07f4cebe5780f1039541d989e64b70eccc5b4eb5) and renamed the internal `peer` to `peer_ref` to avoid a conflict